### DISCUSSION TITLE
having problem in 06-166 loading custom image

### DISCUSSION THREAD

**Discussion by MrinalKB** [mrdbourke/pytorch-deep-learning#478] [Q&A]
`import torchvision`
`custom_image_uint8 = torchvision.io.read_image(str(custom_image_path))`
`custom_image_uint8`

error - Unsupported image file. Only jpeg and png are currently supported.

`custom_image_path` -  PosixPath('data/04-pizza-dad.jpeg')

just as instructed in the video yet getting it wrong

**Comment by AxisMeetsWorld** [comment]
I noticed the same thing. The only thing I can add that might provide further insights is I tried right-clicking on the file and selected "copy path". Here is what I got: `/content/data/04-pizza-dad.jpeg` and I tried adding the **/content** on the beginning. That didn't work either. I also noticed that the jpeg  file didn't open when I tried to open it in Colab, and I had to copy and paste in the exact raw file extension from the book version [here](https://www.learnpytorch.io/04_pytorch_custom_datasets/#10-compare-model-results). That still didn't seem to allow the request.get() to pull up an image. So, I have the feeling there is something wrong with the current raw file @mrdbourke

**Comment by AxisMeetsWorld** [comment]
Hi @MrinalKB and @mrdbourke,
          I think I figured it out. At least with mine, I had single quotes in the requests.get("https://raw.githubusercontent.com/mrdbourke/pytorch-deep-learning/main/images/04-pizza-dad.jpeg") call. When I changed those to double quotes, it worked. The error message isn't real helpful here, but that was the issue. The image came up when I popped that in there. @MrinalKB try that and see if it works for you. Also, @mrdbourke, if we don't see that option to get the raw file from the image directory, how do we get that? Hopefully this solves our problems.

**Comment by mrdbourke** [comment]
Hi @AxisMeetsWorld and @MrinalKB ,

If you're having trouble with displaying the image, see the following code snippet:

```python
# Download custom image
import requests
from pathlib import Path

# Setup path to data folder
data_path = Path("data/")
data_path.mkdir(exist_ok=True)

# Setup custom image path
custom_image_path = data_path / "04-pizza-dad.jpeg"

# Download the image if it doesn't already exist
if not custom_image_path.is_file():
    with open(custom_image_path, "wb") as f:
        # When downloading from GitHub, need to use the "raw" file link
        request = requests.get("https://raw.githubusercontent.com/mrdbourke/pytorch-deep-learning/main/images/04-pizza-dad.jpeg")
        print(f"Downloading {custom_image_path}...")
        f.write(request.content)
else:
    print(f"{custom_image_path} already exists, skipping download.")

# Display image
from IPython.display import Image, display

display(Image(custom_image_path))
```

I can confirm it works in Google Colab, see this attached notebook: https://colab.research.google.com/drive/1eLCCSP38zf6EEgeFI-cdNmrgKwZfSZg8?usp=sharing 

--- 


## Getting the raw file from the image directory

Since GitHub recently changed their design, the "raw" option isn't a button anymore.

So you can now just get the link via going to image and right-clicking and then selecting "Copy Image Address" option.

Steps:
1. Go to original image (e.g. https://github.com/mrdbourke/pytorch-deep-learning/blob/main/images/04-pizza-dad.jpeg) 
2. Right click
3. Select "Copy Image Address" (should end with 'raw?=true', e.g. https://github.com/mrdbourke/pytorch-deep-learning/blob/main/images/04-pizza-dad.jpeg?raw=true)

<img width="1499" alt="pizza-dad-copy-image-address" src="https://github.com/mrdbourke/pytorch-deep-learning/assets/16750345/0ce3e9dc-9666-41f0-baf9-d85b165acf52">

See the new code snippet:

```python
# You can get this by right clicking on the image at: https://github.com/mrdbourke/pytorch-deep-learning/blob/main/images/04-pizza-dad.jpeg 
# And selecting "copy image address"
image_address = "https://github.com/mrdbourke/pytorch-deep-learning/blob/main/images/04-pizza-dad.jpeg?raw=true"

# Download custom image
import requests
from pathlib import Path

# Setup path to data folder
data_path = Path("data/")
data_path.mkdir(exist_ok=True)

# Setup custom image path
custom_image_path = data_path / "04-pizza-dad.jpeg"

# Download the image if it doesn't already exist
if not custom_image_path.is_file():
    with open(custom_image_path, "wb") as f:
        # When downloading from GitHub, need to use the "raw" file link
        request = requests.get(image_address) ### <- CHANGED
        print(f"Downloading {custom_image_path}...")
        f.write(request.content)
else:
    print(f"{custom_image_path} already exists, skipping download.")

# Display image
from IPython.display import Image, display

display(Image(custom_image_path))
```
